From a2de02ce78a20b775a76c400db56303f4cc529e1 Mon Sep 17 00:00:00 2001
From: "Deng, Bing" <bing.deng@intel.com>
Date: Fri, 26 Mar 2021 14:28:58 +0800
Subject: [PATCH] Add
 talk/owt/patches/0017-Support-OMX.Intel.hw_ve.h264-for-android-app.patch.

1. Add talk/owt/patches/0017-Support-OMX.Intel.hw_ve.h264-for-android-app.patch.
2. Compile x86_64 ABI.

Signed-off-by: Deng, Bing <bing.deng@intel.com>
---
 scripts/build_android.py                      |  4 +--
 scripts/prepare_dev.py                        |  3 ++-
 .../0016-Use-AToU-to-print-trace.patch        |  6 ++---
 ...OMX.Intel.hw_ve.h264-for-android-app.patch | 25 +++++++++++++++++++
 4 files changed, 32 insertions(+), 6 deletions(-)
 create mode 100644 talk/owt/patches/0017-Support-OMX.Intel.hw_ve.h264-for-android-app.patch

diff --git a/scripts/build_android.py b/scripts/build_android.py
index 23db934..e9e6ca3 100755
--- a/scripts/build_android.py
+++ b/scripts/build_android.py
@@ -71,8 +71,8 @@ if __name__ == '__main__':
     parser = argparse.ArgumentParser()
 
     #build libs for all platforms by default
-    parser.add_argument('--arch', default = 'arm,arm64,x86', dest = 'target_arch',
-        choices = ['arm', 'arm64', 'x86'],
+    parser.add_argument('--arch', default = 'arm,arm64,x86,x64', dest = 'target_arch',
+        choices = ['arm', 'arm64', 'x86', 'x64'],
         help = 'Target architecture(s) to be built, all arch by default.')
 
     #build release version by default
diff --git a/scripts/prepare_dev.py b/scripts/prepare_dev.py
index e51061d..1440822 100644
--- a/scripts/prepare_dev.py
+++ b/scripts/prepare_dev.py
@@ -44,7 +44,8 @@ patchList = [
     ('0013-Remove-unused-gni-for-av1-build.patch', THIRD_PARTY_PATH),
     ('0014-Fix-missing-ffmpeg-configure-item-for-msvc-build.patch', FFMPEG_PATH),
     ('0015-Adopt-SingletonSurfaceView.patch', WEBRTC_PATH),
-    ('0016-Use-AToU-to-print-trace.patch', WEBRTC_PATH)
+    ('0016-Use-AToU-to-print-trace.patch', WEBRTC_PATH),
+    ('0017-Support-OMX.Intel.hw_ve.h264-for-android-app.patch', WEBRTC_PATH)
 ]
 
 def _patch(ignoreFailures=False):
diff --git a/talk/owt/patches/0016-Use-AToU-to-print-trace.patch b/talk/owt/patches/0016-Use-AToU-to-print-trace.patch
index 01f8175..70a82af 100644
--- a/talk/owt/patches/0016-Use-AToU-to-print-trace.patch
+++ b/talk/owt/patches/0016-Use-AToU-to-print-trace.patch
@@ -1,7 +1,7 @@
-From 95fd763d1497dbaac9042dd1386aa2bda8fb24f2 Mon Sep 17 00:00:00 2001
+From 26577b8297e1c5b22929c56450c0cfb1338388c4 Mon Sep 17 00:00:00 2001
 From: yingzhex <yingzhenx.li@intel.com>
 Date: Thu, 25 Mar 2021 14:35:59 +0800
-Subject: [PATCH] Use AToU to print trace
+Subject: [PATCH 1/2] Use AToU to print trace
 
 Signed-off-by: yingzhex <yingzhenx.li@intel.com>
 ---
@@ -901,5 +901,5 @@ index 0000000000..d8276f2954
 +
 +#endif /* !_STDATOMIC_H_ */
 -- 
-2.17.1
+2.25.1
 
diff --git a/talk/owt/patches/0017-Support-OMX.Intel.hw_ve.h264-for-android-app.patch b/talk/owt/patches/0017-Support-OMX.Intel.hw_ve.h264-for-android-app.patch
new file mode 100644
index 0000000..bb9cb38
--- /dev/null
+++ b/talk/owt/patches/0017-Support-OMX.Intel.hw_ve.h264-for-android-app.patch
@@ -0,0 +1,25 @@
+From a569550ae6c9a0ae35625af33e0166152bb8c8c8 Mon Sep 17 00:00:00 2001
+From: Kai Liu <kai1.liu@intel.com>
+Date: Wed, 24 Mar 2021 10:10:12 +0800
+Subject: [PATCH] Support OMX.Intel.hw_ve.h264 for android app
+
+Signed-off-by: Kai Liu <kai1.liu@intel.com>
+---
+ sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java b/sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java
+index 306c29df70..766718e7c5 100644
+--- a/sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java
++++ b/sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java
+@@ -251,6 +251,7 @@ public class HardwareVideoEncoderFactory implements VideoEncoderFactory {
+                && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
+         || (name.startsWith(HISI_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+         || (name.startsWith(IMG_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
++        || (name.startsWith(INTEL_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+         || vcp.isExtraHardwareSupported(name, "video/avc", vcp.parseWithTag(vcp.loadWithDom(extraMediaCodecFile), "Encoders"));
+   }
+ 
+-- 
+2.25.1
+
-- 
2.25.1

